### PR TITLE
Allow Twig v3+ and upgrade UrlBuilderExtension

### PR DESCRIPTION
**Pull request by ambroisemaupate**
- Removed composer.lock (type: library)
- Allowed Twig v2 and v3
- Removed old Twig classes usage
- Fixed some phpstan errors about strict-typing on string functions
- Allowed Travis to test on latest PHP versions

### CODE REVIEW

File: src/Twig/UrlBuilderExtension.php [hunk 1]
```diff
@@ -34,12 +36,16 @@ public function __construct(Builder $builder)
     public function getFilters()
     {
         return [
-            new \Twig_SimpleFilter('imageUrl', [$this, 'imageUrl']),
+            new TwigFilter('imageUrl', [$this, 'imageUrl']),
         ];
     }
 
     /**
-     * buildUrl
```
Comment (jdeniau) [inline src/Twig/UrlBuilderExtension.php:42]: Useless as it is already defined in PHP directly.

Is it a phpstorm issue or something like that ?

File: .gitignore [hunk 2]
```diff
@@ -5,4 +5,4 @@ clover.xml
 
 # Commit your application's lock file https://getcomposer.org/doc/01-basic-usage.md#commit-your-composer-lock-file-to-version-control
 # You may choose to ignore a library lock file http://getcomposer.org/doc/02-libraries.md#lock-file
-# composer.lock
```
Comment (jdeniau) [inline .gitignore]: Can you reset the composer.lock file. This way, our phpstan version and stuff stay in a "managed" version.

**Review by jdeniau — CHANGES_REQUESTED** [review]

File: src/Twig/UrlBuilderExtension.php (hunk 1 above)
Comment (ambroisemaupate) [reply src/Twig/UrlBuilderExtension.php:42]: I removed it because it was not current method name. I wrote it back with `imageUrl`

**Comment by ambroisemaupate** [comment]
Travis fails on php7.1 because of `composer.lock` generated on php7.4, it tries to install Twig 3.0.5 to respect `lock` but Twig 3.0.5 needs php7.2 min.

**Comment by jdeniau** [comment]
bumped as `2.0.0`.

I dropped support for php 7.1 (unsupported for a year now). I added static types whenever possible.

See [the changelog](https://github.com/mapado/image-url-builder/blob/master/CHANGELOG.md#200)

**Comment by ambroisemaupate** [comment]
Thanks @jdeniau !

**Pull request merged**